src/InputWithlabel.jsx

```diff
@@ -0,0 +1,26 @@
+/* eslint-disable no-unused-vars */
+import React, { useEffect, useRef } from 'react';
+
+const InputWithLabel = ({ value, onChange, id, type = "text", children }) => {
+    const inputRef = useRef(null);
+
+    useEffect(() => {
+        inputRef.current.focus();
+      }, []);
```

Comment:
You need to remove the empty dependency array. Without it, the input will focus both after adding a task to the list and after deleting one. Currently, with the empty dependency array, your useEffect only triggers on the initial render.

```
    useEffect(() => {
        inputRef.current.focus();
      });
```